main.go

```diff
@@ -114,128 +240,107 @@ func main() {
-		nameCleaned := strings.TrimSpace(string(nameRaw))
+func detach(parentSessionID string) {
+	net.Dial("unix", elaborateSessionInfo("", parentSessionID).detachPath)
+}
 
-		if nameCleaned == sessionName {
-			return id, nil
-		}
+func elaborateSessionInfo(name string, uuid string) SessionInfo {
+	dirPath := path.Join(threemuxDir, uuid)
+	return SessionInfo{
+		name:           name,
+		uuid:           uuid,
+		path:           dirPath,
+		fdPath:         path.Join(dirPath, "fd.sock"),
+		killClientPath: path.Join(dirPath, "kill-client.sock"),
+		killServerPath: path.Join(dirPath, "kill-server.sock"),
+		detachPath:     path.Join(dirPath, "detach-server.sock"),
+		resizePath:     path.Join(dirPath, "resize.sock"),
+		logsPath:       path.Join(dirPath, "logs-server.txt"),
 	}
-
-	return "", fmt.Errorf("Could not find session: `%s`", sessionName)
 }
 
-func launchServer(sessionName string) (sessionID string) {
-	os.MkdirAll(threemuxDir, 0777)
-
-	id := 0
-	for {
-		dir := path.Join(threemuxDir, strconv.Itoa(id))
-		if _, err := os.Stat(dir); os.IsNotExist(err) {
-			break
-		}
-		id++
+func initializeSession(sessionName string) SessionInfo {
+	sessionUUIDObj, err := uuid.NewV4()
+	if err != nil {
+		fmt.Println("Session creation failed. Error while generating session UUID:", err)
+		fmt.Printf("Please report this to %s\n", BUG_REPORT_URL)
+		os.Exit(1)
 	}
-	sessionID = strconv.Itoa(id)
+	sessionUUID := sessionUUIDObj.String()
 
-	dir := path.Join(threemuxDir, sessionID)
-	os.MkdirAll(dir, 0755)                                              // FIXME perms
-	ioutil.WriteFile(path.Join(dir, "name"), []byte(sessionName), 0777) // FIXME perms
+	sessionPath := path.Join(threemuxDir, sessionUUID)
 
-	readySocket, err := net.Listen("unix", path.Join(dir, "booted.sock"))
+	err = os.Mkdir(sessionPath, 0700)
 	if err != nil {
-		panic(err)
+		fmt.Println("Session creation failed. Error while creating session data directory:", err)
+		fmt.Printf("Please report this to %s\n", BUG_REPORT_URL)
+		os.Exit(1)
 	}
 
-	args := []string{"3mux", "_serve-id", sessionID}
-	if *writeLogs {
-		args = append(args, "--log")
+	nameFilePath := path.Join(sessionPath, "name")
+	err = ioutil.WriteFile(nameFilePath, []byte(sessionName), 0600)
+	if err != nil {
+		fmt.Println("Session creation failed. Error while recording session metadata:", err)
+		err = os.Remove(sessionPath)
+		if err != nil {
+			fmt.Printf(
+				"Session clean-up failed. Error while removing directory `%s`: %s",
+				sessionPath, err.Error(),
+			)
+		}
+		fmt.Printf("Please report this to %s\n", BUG_REPORT_URL)
+		os.Exit(1)
 	}
 
-	context := &daemon.Context{
-		// PidFileName: "sample.pid",
-		Args: args,
-	}
-	_, err = context.Reborn()
+	return elaborateSessionInfo(sessionName, sessionUUID)
+}
+
+func findSession(sessionName string) (sessionInfo SessionInfo, found bool, err error) {
+	children, err := ioutil.ReadDir(threemuxDir)
 	if err != nil {
-		panic(err)
+		return SessionInfo{}, false, err
```

Comment:
If the return type was a pointer to a `SessionInfo` type, you could save memory by using `nil` instead of `SessionInfo{}`